ec/ecresource/projectresource/read_test.go

```diff
@@ -0,0 +1,187 @@
+package projectresource
+
+import (
+	"context"
+	"testing"
+
+	"github.com/elastic/terraform-provider-ec/ec/internal/gen/serverless/resource_elasticsearch_project"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/tfsdk"
+	"github.com/hashicorp/terraform-plugin-framework/types/basetypes"
+	"github.com/hashicorp/terraform-plugin-go/tftypes"
+	"github.com/stretchr/testify/require"
+	"go.uber.org/mock/gomock"
+)
+
+func TestRead(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	type testData struct {
+		modelHandler        modelHandler[resource_elasticsearch_project.ElasticsearchProjectModel]
+		api                 api[resource_elasticsearch_project.ElasticsearchProjectModel]
+		req                 resource.ReadRequest
+		expectedDiags       diag.Diagnostics
+		expectStateMutation bool
+		expectNullState     bool
+		expectedId          *string
+	}
+	tests := []struct {
+		name     string
+		testData func(context.Context) testData
+	}{
+		{
+			name: "should fail if reading the tf model errors",
+			testData: func(ctx context.Context) testData {
+				req := resource.ReadRequest{
```

Comment:
The similar consideration as for create - `req` isn't needed IMO.

Reply:
Similarly to the Create test. Let's make a decision there and I'll make sure it's consistent across all the tests :)